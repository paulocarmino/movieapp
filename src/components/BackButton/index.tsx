import { IoMdArrowBack } from 'react-icons/io'

export type BackButtonProps = {
  size: number
  action?: () => void
}

const BackButton = ({ size, action = () => { } }: BackButtonProps) => {
  return (
    <div data-testid="back-icon" onClick={() => action()}>
      <IoMdArrowBack className='hover:text-gray-600 cursor-pointer' size={size} />
    </div>
  )
}

export default BackButton
