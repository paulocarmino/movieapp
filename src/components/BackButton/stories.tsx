import { Story, Meta } from '@storybook/react'
import BackButton, { BackButtonProps } from '.'

export default {
  title: 'Back Button',
  component: BackButton,
  parameters: {
    backgrounds: {
      default: 'dark'
    }
  },
  argTypes: {
    children: {
      type: 'string'
    }
  }
} as Meta

export const Default: Story<BackButtonProps> = (args) => <BackButton {...args} />

Default.args = {
  size: 24
}
