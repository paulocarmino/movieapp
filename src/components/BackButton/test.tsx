import { fireEvent, render, screen } from '@testing-library/react'

import BackButton from '.'

describe('<Badge />', () => {
  it('should render the component', () => {
    const { container } = render(<BackButton size={24} />)

    expect(screen.getByTestId('back-icon')).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should trigger action when passed', () => {
    const actionMock = jest.fn();
    const { getByTestId } = render(<BackButton size={24} action={actionMock} />)

    fireEvent.click(getByTestId('back-icon'))
    expect(actionMock).toBeCalled()
  })
})
