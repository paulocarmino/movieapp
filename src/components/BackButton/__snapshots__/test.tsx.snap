// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge /> should render the component 1`] = `
<div
  data-testid="back-icon"
>
  <svg
    class="hover:text-gray-600 cursor-pointer"
    fill="currentColor"
    height="24"
    stroke="currentColor"
    stroke-width="0"
    viewBox="0 0 512 512"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M427 234.625H167.296l119.702-119.702L256 85 85 256l171 171 29.922-29.924-118.626-119.701H427v-42.75z"
    />
  </svg>
</div>
`;
